src/VirtoCommerce.CartModule.Data/Model/DiscountEntityComparer.cs

```diff
@@ -11,7 +11,7 @@ public bool Equals(DiscountEntity x, DiscountEntity y)
             if (x != null && y != null)
             {
                 equals = x.PromotionId == y.PromotionId &&
-                         x.PromotionDescription == y.PromotionDescription &&
```

Comment:
Don't remove PromotionDescription and put it after PromotionName.
The same for GetHashCode().

Reply:
fixed